### PR TITLE
JENKINS-54662 JCasC support for usage statistics

### PR DESCRIPTION
**Pull request by timja**
See [JENKINS-54662](https://issues.jenkins-ci.org/browse/JENKINS-54662).

Example configuration:
```yaml
jenkins:
  noUsageStatistics: true
```

### Proposed changelog entries

* Entry 1: JENKINS-54662, configuration-as-code plugin support for disabling usage statistics
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 1]
```diff
@@ -1373,11 +1373,30 @@ public UpdateCenter getUpdateCenter() {
...
+        return !noUsageStatistics;
     }
 
-    public void setNoUsageStatistics(Boolean noUsageStatistics) throws IOException {
+    /**
+     * Sets the noUsageStatistics flag
+     *
+     * @since TODO
+     */
+    public void setNoUsageStatistics(boolean noUsageStatistics) throws IOException {
```
Comment (timja) [inline core/src/main/java/jenkins/model/Jenkins.java]: type change seems safe:
https://github.com/search?q=org%3Ajenkinsci+setNoUsageStatistics&type=Code

**Review by jetersen — APPROVED** [review]
Seems reasonable

**Review by rsandell — APPROVED** [review]

**Review by alecharp — APPROVED** [review]

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 2]
```diff
@@ -1373,10 +1373,29 @@ public UpdateCenter getUpdateCenter() {
...
+     * @since TODO
+     */
     public boolean isUsageStatisticsCollected() {
         return noUsageStatistics==null || !noUsageStatistics;
     }
 
+    /**
+     * Sets the noUsageStatistics flag
+     *
+     * @since TODO
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/Jenkins.java]: It was introduced in 2011 or earlier, so it is not a big deal.  All reasonable  versions will be compatible https://github.com/jenkinsci/jenkins/commit/70c10658148c3eab3b4c4122705e1ed3a12e193d

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 3]
```diff
@@ -1373,10 +1373,29 @@ public UpdateCenter getUpdateCenter() {
         return updateCenter;
     }
 
+    /**
+     * If usage statistics has been disabled
+     *
+     * @since TODO
+     */
+    public Boolean isNoUsageStatistics() {
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/Jenkins.java]: Just to make it explicit why the method is different

```suggestion
    @CheckForNull
    public Boolean isNoUsageStatistics() {
```

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 4]
```diff
@@ -1373,10 +1373,29 @@ public UpdateCenter getUpdateCenter() {
...
+        return noUsageStatistics;
+    }
+
+    /**
+     * If usage statistics are being collected
+     *
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/Jenkins.java]:
```suggestion
     *
     * @return {@code true} if usage statistics should be collected.
     *                Defaults to {@code true} when {@link #noUsageStatistics} is not set.
```

**Review by oleg-nenashev — APPROVED** [review]
The code itself looks good to me. Added some suggestions for Javadoc so that two methods can be distinguished easily.

**Comment by timja** [comment]
This is ready-for-merge, let's merge it in 24 hours if no negative feedback

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 5]
```diff
@@ -1373,10 +1373,32 @@ public UpdateCenter getUpdateCenter() {
...
+     * @return {@code true} if usage statistics should be collected.
+     *                Defaults to {@code true} when {@link #noUsageStatistics} is not set.
+     * @since TODO
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/Jenkins.java:1391]: This method has existed forever. Why add `@since`?

Comment (daniel-beck) [reply core/src/main/java/jenkins/model/Jenkins.java:1391]: https://github.com/jenkinsci/jenkins/pull/4706

**Pull request merged**